### ISSUE TITLE
refactor: Rewrite nn.ch using tensor primitives

### ISSUE THREAD

**Issue by petrucristescu** [petrucristescu/churing#75]
## Summary

Replace the linked-list-based ops in `nn.ch` with the new tensor primitives from #74. The network weights, biases, activations, and gradients all become `Tensor` instead of `[[Float]]` / `[Float]`.

## Changes

`nn.ch` — swap every list op for its tensor equivalent:

```
# before
@z1 (vecAdd (matVecMul w1 input) b1)
@a1 (map relu z1)

# after
@z1 (tensorAdd (tensorMatVecMul w1 input) b1)
@a1 (tensorRelu z1)
```

Full mapping:

| Before | After |
|---|---|
| `matVecMul` | `tensorMatVecMul` |
| `vecAdd / vecSub / vecMul / vecScale` | `tensorAdd / tensorSub / tensorMul / tensorScale` |
| `matAdd / matScale / matTranspose` | `tensorAdd / tensorScale / tensorTranspose` |
| `outerProduct` | `tensorOuterProduct` |
| `softmax` | `tensorSoftmax` |
| `map relu` / `map reluDeriv` | `tensorRelu` / `tensorReluDeriv` |
| `argmax` | `tensorArgmax` |
| `matRandom` / `vecZeros` | `tensorRandom` / `tensorCreate` |
| `oneHot` | `tensorFromList` of a sparse list |

## Boundary helpers

Add `listToTensor` / `tensorToList` conversion helpers so existing boundary code (pgm.ch reading pixels as lists, JSON loading weights) continues to work.

## Benchmark

After this change, re-run `examples/benchmark.ch` and compare against the pre-tensor baseline. Target: full digit NN training under 60 seconds.

## Depends on

#74 (CPU tensor math ops)
